commons/src/main/java/io/aiven/kafka/tieredstorage/commons/security/EncryptionKeyProvider.java

```diff
@@ -69,6 +71,15 @@ public SecretKey createKey() {
         return aesKeyGenerator.generateKey();
     }
 
+    public EncryptionKeyAndAAD createKeyAndAAD() {
```

Comment:
Should we rename the returned class to `SecretKeyAndAAD` instead to keep it aligned with `createKey(): SecretKey`?

Reply:
> Should we have a test for it also?

Not sure. What to test, that it doesn't return `null`?

Reply:
> Should we rename the returned class to `SecretKeyAndAAD` instead to keep it aligned with `createKey(): SecretKey`?

Unification of these two terms throughout the code base is indeed needed. I personally would prefer "encryption" over "secret". But this probably goes beyond the scope of this PR. I've made https://github.com/aiven/tiered-storage-for-apache-kafka/issues/166 for this.